aptos-move/aptos-vm/src/move_vm_ext/vm.rs

```diff
@@ -67,16 +69,9 @@ impl MoveVmExt {
             type_byte_cost = 1;
         }
 
-        Ok(Self {
-            inner: MoveVM::new_with_config(
-                aptos_natives(
-                    gas_feature_version,
-                    native_gas_params,
-                    misc_gas_params,
-                    timed_features,
-                    features.clone(),
-                ),
-                VMConfig {
+        if let Some(builder) = builder_option {
```

Comment:
Similarly, let's get rid of the duplicate code here by

1. Moving the actual implementation into the following function
```
fn new_impl<F>(
        native_gas_params: NativeGasParameters,
        misc_gas_params: MiscGasParameters,
        gas_feature_version: u64,
        chain_id: u8,
        features: Features,
        timed_features: TimedFeatures,
        gas_hook: Option<F>,
    ) -> VMResult<Self> 
    where
        F: Fn(DynamicExpression)
    {
        ..
    }
}
```

2. Make `pub fn new` call `new_impl` with gas_hook set to `None`

3. Introduce a new interface
```
pub fn new_with_gas_hook(.., gas_hook) -> VMResult<Self> {
    new_impl(.., Some(gas_hook))
}
```

Reply:
Addressed the refactor, it should be good to go now.